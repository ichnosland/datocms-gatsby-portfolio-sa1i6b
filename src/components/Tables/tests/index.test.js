import React from 'react';
import { shallow } from 'enzyme';

import { Table, Tr, Td } from '../index';

describe('<Table />', () => {
  it('should render a <table> tag', () => {
    const renderedComponent = shallow(<Table />);
    expect(renderedComponent.type()).toEqual('table');
  });

  it('should render its css w/ props bgcolor', () => {
    const bgcolor = 'red';
    const renderedComponent = shallow(<Table bgcolor={bgcolor} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props fixed', () => {
    const renderedComponent = shallow(<Table fixed />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props neutral', () => {
    const renderedComponent = shallow(<Table neutral />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props pricing', () => {
    const renderedComponent = shallow(<Table pricing />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<Tr />', () => {
  it('should render a <tr> tag', () => {
    const renderedComponent = shallow(<Tr />);
    expect(renderedComponent.type()).toEqual('tr');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Tr id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(
      <Tr>{children}</Tr>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });

  it('should render its css w/ props Br', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Tr Br={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props Bl', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Tr Bl={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props Bt', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Tr Bt={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props Bb', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Tr Bb={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props fullBg w/o further props', () => {
    const renderedComponent = shallow(<Tr fullBg />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props fullBg & BgÂ & color', () => {
    const color = 'yellow';
    const Bg = 'red';
    const renderedComponent = shallow(<Tr fullBg Bg={Bg} color={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<Td />', () => {
  it('should render a <td> tag', () => {
    const renderedComponent = shallow(<Td />);
    expect(renderedComponent.type()).toEqual('td');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Td id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(
      <Td>{children}</Td>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });

  it('should render its css w/ props bgcolor', () => {
    const color = '1px solid red';
    const renderedComponent = shallow(<Td bgcolor={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props Br', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Td Br={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props Bl', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Td Bl={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props Bt', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Td Bt={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props Bb', () => {
    const color = '1px solid blue';
    const renderedComponent = shallow(<Td Bb={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
