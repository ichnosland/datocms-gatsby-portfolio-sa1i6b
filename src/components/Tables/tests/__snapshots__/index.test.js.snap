// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render its css w/ props bgcolor 1`] = `
.c0 {
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  border-width: 1px;
  border-style: solid;
  background-color: red;
}

.c0 strong {
  font-weight: 700;
}

@media (max-width:30em) {
  .c0 {
    font-size: 12px;
  }
}

<table
  className="c0"
/>
`;

exports[`<Table /> should render its css w/ props fixed 1`] = `
.c0 {
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  border-width: 1px;
  border-style: solid;
  width: 100%;
  table-layout: fixed;
}

.c0 strong {
  font-weight: 700;
}

@media (max-width:30em) {
  .c0 {
    font-size: 12px;
  }
}

<table
  className="c0"
/>
`;

exports[`<Table /> should render its css w/ props neutral 1`] = `
.c0 {
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0,133,178,1);
}

.c0 td {
  vertical-align: top;
  padding: 16px;
}

.c0 strong {
  font-weight: 700;
}

@media (max-width:30em) {
  .c0 {
    font-size: 12px;
  }
}

@media (max-width:41.6875em) {
  .c0 td {
    padding: 12px;
  }

  .c0 td ul {
    padding: 10px;
  }
}

<table
  className="c0"
/>
`;

exports[`<Table /> should render its css w/ props pricing 1`] = `
.c0 {
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  border-width: 1px;
  border-style: solid;
  color: white;
  border-color: white;
}

.c0 strong {
  font-weight: 700;
}

@media (max-width:30em) {
  .c0 {
    font-size: 12px;
  }
}

<table
  className="c0"
/>
`;

exports[`<Td /> should render its css w/ props Bb 1`] = `
.c0 {
  padding: 14px 5px;
  overflow: hidden;
  word-break: normal;
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: 1px solid blue;
}

<td
  className="c0"
/>
`;

exports[`<Td /> should render its css w/ props Bl 1`] = `
.c0 {
  padding: 14px 5px;
  overflow: hidden;
  word-break: normal;
  border-top: none;
  border-right: none;
  border-left: 1px solid blue;
  border-bottom: none;
}

<td
  className="c0"
/>
`;

exports[`<Td /> should render its css w/ props Br 1`] = `
.c0 {
  padding: 14px 5px;
  overflow: hidden;
  word-break: normal;
  border-top: none;
  border-right: 1px solid blue;
  border-left: none;
  border-bottom: none;
}

<td
  className="c0"
/>
`;

exports[`<Td /> should render its css w/ props Bt 1`] = `
.c0 {
  padding: 14px 5px;
  overflow: hidden;
  word-break: normal;
  border-top: 1px solid blue;
  border-right: none;
  border-left: none;
  border-bottom: none;
}

<td
  className="c0"
/>
`;

exports[`<Td /> should render its css w/ props bgcolor 1`] = `
.c0 {
  padding: 14px 5px;
  overflow: hidden;
  word-break: normal;
  background-color: 1px solid red;
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: none;
}

<td
  className="c0"
/>
`;

exports[`<Tr /> should render its css w/ props Bb 1`] = `
.c0 {
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: 1px solid blue;
}

<tr
  className="c0"
/>
`;

exports[`<Tr /> should render its css w/ props Bl 1`] = `
.c0 {
  border-top: none;
  border-right: none;
  border-left: 1px solid blue;
  border-bottom: none;
}

<tr
  className="c0"
/>
`;

exports[`<Tr /> should render its css w/ props Br 1`] = `
.c0 {
  border-top: none;
  border-right: 1px solid blue;
  border-left: none;
  border-bottom: none;
}

<tr
  className="c0"
/>
`;

exports[`<Tr /> should render its css w/ props Bt 1`] = `
.c0 {
  border-top: 1px solid blue;
  border-right: none;
  border-left: none;
  border-bottom: none;
}

<tr
  className="c0"
/>
`;

exports[`<Tr /> should render its css w/ props fullBg & BgÂ & color 1`] = `
.c0 {
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: none;
  background-color: red;
  color: yellow;
}

<tr
  className="c0"
  color="yellow"
/>
`;

exports[`<Tr /> should render its css w/ props fullBg w/o further props 1`] = `
.c0 {
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: none;
  background-color: white;
  color: #0085b2;
}

<tr
  className="c0"
/>
`;
